### ISSUE TITLE
test should return 409 or 204 when statement ID already exists on PUT, checks POST instead

### ISSUE THREAD

**Issue by bscSCORM** [adlnet/lrs-conformance-test-suite#136]
The test "should return 409 or 204 when statement ID already exists on PUT"

Does both a `PUT` and a `POST`. We respond `204` to the `PUT`, and then `200` to the `POST`, at which point we get the error: `Error: Received status code: 200`

However, I see nothing in the spec to indicate that `200` is wrong for `POST` (`409` would presumably also be OK).

```
{
    "name": "console",
    "owner": null,
    "flags": {
        "endpoint": "http://localhost:64619/ScormEngineInterface/TCAPI/default",
        "basicAuth": true,
        "authUser": "scorm",
        "grep": "should return 409 or 204 when statement ID already exists on PUT"
    },
    "options": {},
    "rollupRule": "mustPassAll",
    "uuid": "3c92bb86-8651-43ea-8cd6-2352a13bb723",
    "startTime": 1485627716075,
    "endTime": 1485627716325,
    "duration": 250,
    "state": "finished",
    "summary": {
        "total": 1,
        "passed": 0,
        "failed": 1,
        "version": "1.0.3.7"
    },
    "log": {
        "title": "",
        "name": "",
        "requirement": "",
        "log": "REQUEST SUPERREQUEST\n_______________________________________\nPOST /ScormEngineInterface/TCAPI/default/statements HTTP/1.1\r\nX-Experience-API-Version: 1.0.3\r\nAuthorization: Basic c2Nvcm06c2Nvcm0yMDA0\r\nhost: localhost:64619\r\naccept: application/json\r\ncontent-type: application/json\r\ncontent-length: 324\r\nConnection: close\r\n\r\n{\"actor\":{\"objectType\":\"Agent\",\"name\":\"xAPI mbox\",\"mbox\":\"mailto:xapi@adlnet.gov\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/attended\",\"display\":{\"en-GB\":\"attended\",\"en-US\":\"attended\"}},\"object\":{\"objectType\":\"Activity\",\"id\":\"http://www.example.com/meetings/occurances/34534\"},\"id\":\"3b1b670b-f46a-402e-97bb-5a9550365038\"}\n\nRESPONSE SUPERREQUEST\n_______________________________________\nHTTP/1.1 200 OK\nserver: ASP.NET Development Server/11.0.0.0\ndate: Sat, 28 Jan 2017 18:21:56 GMT\nx-aspnet-version: 4.0.30319\nx-experience-api-version: 1.0.3\naccess-control-allow-origin: *\naccess-control-allow-methods: HEAD,GET,POST,PUT,DELETE\naccess-control-allow-headers: Content-Type,Content-Length,Authorization,If-Match,If-None-Match,X-Experience-API-Version,X-Experience-API-Consistent-Through\naccess-control-expose-headers: ETag,Last-Modified,Cache-Control,Content-Type,Content-Length,WWW-Authenticate,X-Experience-API-Version,X-Experience-API-Consistent-Through\ntransfer-encoding: chunked\ncache-control: private\ncontent-type: application/json; charset=UTF-8\nconnection: Close\n\n3b1b670b-f46a-402e-97bb-5a9550365038\n=======================================\nREQUEST SUPERREQUEST\n_______________________________________\nGET /ScormEngineInterface/TCAPI/default/statements?statementId=3b1b670b-f46a-402e-97bb-5a9550365038 HTTP/1.1\r\nX-Experience-API-Version: 1.0.3\r\nAuthorization: Basic c2Nvcm06c2Nvcm0yMDA0\r\nhost: localhost:64619\r\nConnection: close\r\n\r\nRESPONSE SUPERREQUEST\n_______________________________________\nHTTP/1.1 200 OK\nserver: ASP.NET Development Server/11.0.0.0\ndate: Sat, 28 Jan 2017 18:21:56 GMT\nx-aspnet-version: 4.0.30319\nx-experience-api-version: 1.0.3\naccess-control-allow-origin: *\naccess-control-allow-methods: HEAD,GET,POST,PUT,DELETE\naccess-control-allow-headers: Content-Type,Content-Length,Authorization,If-Match,If-None-Match,X-Experience-API-Version,X-Experience-API-Consistent-Through\naccess-control-expose-headers: ETag,Last-Modified,Cache-Control,Content-Type,Content-Length,WWW-Authenticate,X-Experience-API-Version,X-Experience-API-Consistent-Through\nx-experience-api-consistent-through: 2017-01-28T18:21:55.986Z\ntransfer-encoding: chunked\ncache-control: private\ncontent-type: application/json; charset=UTF-8\nconnection: Close\n\n{\"version\":\"1.0.0\",\"id\":\"3b1b670b-f46a-402e-97bb-5a9550365038\",\"actor\":{\"objectType\":\"Agent\",\"name\":\"xAPI mbox\",\"mbox\":\"mailto:xapi@adlnet.gov\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/attended\",\"display\":{\"en-GB\":\"attended\",\"en-US\":\"attended\"}},\"object\":{\"objectType\":\"Activity\",\"id\":\"http://www.example.com/meetings/occurances/34534\"},\"timestamp\":\"2017-01-28T18:21:56.123Z\",\"stored\":\"2017-01-28T18:21:56.123Z\",\"authority\":{\"objectType\":\"Agent\",\"account\":{\"homePage\":\"http://example.org/bsc/\",\"name\":\"scorm\"}}}\n=======================================\n",
        "status": "failed",
        "tests": [
            {
                "title": "Statement Resource Requirements (Communication 2.1)",
                "name": "Statement Resource Requirements",
                "requirement": [
                    "Communication.md#2.1"
                ],
                "log": "",
                "status": "failed",
                "tests": [
                    {
                        "title": "An LRS's Statement Resource rejects with error code 409 Conflict any Statement with the \"statementID\" parameter equal to a Statement in the LRS already **Implicit** (Communication 2.1.1.s2.b2)",
                        "name": "An LRS's Statement Resource rejects with error code 409 Conflict any Statement with the \"statementID\" parameter equal to a Statement in the LRS already **Implicit**",
                        "requirement": [
                            "Communication.md#2.1.1.s2.b2"
                        ],
                        "log": "",
                        "status": "failed",
                        "tests": [
                            {
                                "title": "should return 409 or 204 when statement ID already exists on PUT",
                                "name": "should return 409 or 204 when statement ID already exists on PUT",
                                "requirement": "",
                                "log": "REQUEST SUPERREQUEST\n_______________________________________\nPUT /ScormEngineInterface/TCAPI/default/statements?statementId=76a7f886-4e97-4163-a47c-00b265a65d0f HTTP/1.1\r\nX-Experience-API-Version: 1.0.3\r\nAuthorization: Basic c2Nvcm06c2Nvcm0yMDA0\r\nhost: localhost:64619\r\naccept: application/json\r\ncontent-type: application/json\r\ncontent-length: 324\r\nConnection: close\r\n\r\n{\"actor\":{\"objectType\":\"Agent\",\"name\":\"xAPI mbox\",\"mbox\":\"mailto:xapi@adlnet.gov\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/attended\",\"display\":{\"en-GB\":\"attended\",\"en-US\":\"attended\"}},\"object\":{\"objectType\":\"Activity\",\"id\":\"http://www.example.com/meetings/occurances/34534\"},\"id\":\"76a7f886-4e97-4163-a47c-00b265a65d0f\"}\n\nRESPONSE SUPERREQUEST\n_______________________________________\nHTTP/1.1 204 No Content\nserver: ASP.NET Development Server/11.0.0.0\ndate: Sat, 28 Jan 2017 18:21:56 GMT\nx-aspnet-version: 4.0.30319\nx-experience-api-version: 1.0.3\naccess-control-allow-origin: *\naccess-control-allow-methods: HEAD,GET,POST,PUT,DELETE\naccess-control-allow-headers: Content-Type,Content-Length,Authorization,If-Match,If-None-Match,X-Experience-API-Version,X-Experience-API-Consistent-Through\naccess-control-expose-headers: ETag,Last-Modified,Cache-Control,Content-Type,Content-Length,WWW-Authenticate,X-Experience-API-Version,X-Experience-API-Consistent-Through\ncache-control: private\nconnection: Close\n\nundefined\n=======================================\nREQUEST SUPERREQUEST\n_______________________________________\nPOST /ScormEngineInterface/TCAPI/default/statements HTTP/1.1\r\nX-Experience-API-Version: 1.0.3\r\nAuthorization: Basic c2Nvcm06c2Nvcm0yMDA0\r\nhost: localhost:64619\r\naccept: application/json\r\ncontent-type: application/json\r\ncontent-length: 324\r\nConnection: close\r\n\r\n{\"actor\":{\"objectType\":\"Agent\",\"name\":\"xAPI mbox\",\"mbox\":\"mailto:xapi@adlnet.gov\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/attended\",\"display\":{\"en-GB\":\"attended\",\"en-US\":\"attended\"}},\"object\":{\"objectType\":\"Activity\",\"id\":\"http://www.example.com/meetings/occurances/34534\"},\"id\":\"76a7f886-4e97-4163-a47c-00b265a65d0f\"}\n\nRESPONSE SUPERREQUEST\n_______________________________________\nHTTP/1.1 200 OK\nserver: ASP.NET Development Server/11.0.0.0\ndate: Sat, 28 Jan 2017 18:21:56 GMT\nx-aspnet-version: 4.0.30319\nx-experience-api-version: 1.0.3\naccess-control-allow-origin: *\naccess-control-allow-methods: HEAD,GET,POST,PUT,DELETE\naccess-control-allow-headers: Content-Type,Content-Length,Authorization,If-Match,If-None-Match,X-Experience-API-Version,X-Experience-API-Consistent-Through\naccess-control-expose-headers: ETag,Last-Modified,Cache-Control,Content-Type,Content-Length,WWW-Authenticate,X-Experience-API-Version,X-Experience-API-Consistent-Through\ntransfer-encoding: chunked\ncache-control: private\ncontent-type: application/json; charset=UTF-8\nconnection: Close\n\n76a7f886-4e97-4163-a47c-00b265a65d0f\n=======================================\n",
                                "status": "failed",
                                "error": "Error: Received status code: 200",
                                "tests": []
                            }
                        ]
                    }
                ]
            }
        ]
    }
}
```

**Comment by bscSCORM** [comment]
Note: I saw this issue as of 2efa228 (not suggesting it was caused by this commit, just noting the latest commit)

**Comment by bscSCORM** [comment]
@cr8onski this is closed, but the reference PR is only merged to `development`, not `master`, shouldn't this stay open until the fix hits `master` ?

**Comment by bscSCORM** [comment]
verified the fix on f735fe8cdce875d1d56d05112062955a343258d3
